### PR TITLE
Support comparison function in().

### LINKED ISSUE #1040: support \in in the WHERE clause
like SQL IN, nGQL should support IN such as:

    WHERE column_name IN (value1,value2,...);

The Set ops (Intersect, Union, Minus) can't work in this condition.

### PR DESCRIPTION
**Pull request by CPWstatic**
I notice that _**in()**_ is a comparison function in Mysql. So I learn from the design and implement it with a function.
The syntax would looks like:
```
YIELD 1 in(1,2,3);
```

Or you may reference it in filter:
```
WHERE 1 in(1,2,3)
```

Definitely, you can reference the input , src and dest ref:
```
WHERE $-.id in(1,2,3)
WHERE $$.team.name in("Spurs","Suns")
WHERE $^.player.name in("Tim Duncan", "Tony Parker")
```

close #1040

### CODE REVIEW

File: src/graph/test/GoTest.cpp [hunk 1]
```diff
@@ -626,5 +626,69 @@ TEST_F(GoTest, NotExistTagProp) {
...
     }
 }
+
+TEST_F(GoTest, inCall) {
+    {
+        cpp2::ExecutionResponse resp;
+        auto &player = players_["Boris Diaw"];
+        auto *fmt = "GO FROM %ld OVER serve "
+                    "WHERE $$.team.name in (\"Hawks\", \"Suns\") "
+                    "YIELD $^.player.name, serve.start_year, serve.end_year, $$.team.name";
```
Comment (whitewum) [inline src/graph/test/GoTest.cpp]: Great

File: src/graph/test/GoTest.cpp [hunk 2]
```diff
@@ -626,5 +626,69 @@ TEST_F(GoTest, NotExistTagProp) {
...
+        std::vector<std::tuple<std::string, int64_t, int64_t, std::string>> expected = {
+            {player.name(), 2003, 2005, "Hawks"},
+            {player.name(), 2005, 2008, "Suns"},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        auto *fmt = "GO FROM %ld OVER like YIELD like._dst AS id"
+                    "| GO FROM $-.id OVER serve WHERE $-.id in(%ld, 123)";
```
Comment (whitewum) [inline src/graph/test/GoTest.cpp]: Great

File: src/graph/test/GoTest.cpp [hunk 3]
```diff
@@ -626,5 +626,69 @@ TEST_F(GoTest, NotExistTagProp) {
...
+            {"serve._dst"}
+        };
+        ASSERT_TRUE(verifyColNames(resp, expectedColNames));
+
+        std::vector<std::tuple<int64_t>> expected = {
+            {teams_["Spurs"].vid()},
+            {teams_["Hornets"].vid()},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
```
Comment (yixinglu) [inline src/graph/test/GoTest.cpp]: Add failed UT for only one parameter in `IN()` function

File: src/graph/test/GoTest.cpp [hunk 4]
```diff
@@ -626,5 +626,69 @@ TEST_F(GoTest, NotExistTagProp) {
         ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
     }
 }
+
+TEST_F(GoTest, inCall) {
+    {
+        cpp2::ExecutionResponse resp;
+        auto &player = players_["Boris Diaw"];
+        auto *fmt = "GO FROM %ld OVER serve "
+                    "WHERE $$.team.name in (\"Hawks\", \"Suns\") "
```
Comment (yixinglu) [inline src/graph/test/GoTest.cpp]: Want `NOT IN` test case

**Comment by yixinglu** [comment]
fix #1040

File: src/graph/test/GoTest.cpp (hunk 4 above)
Comment (whitewum) [reply src/graph/test/GoTest.cpp]: Right.

**Pull request closed without merging**